### DISCUSSION TITLE
A tutorial to integrate other open-source LLMs on HF

### DISCUSSION THREAD

**Discussion by rederyang** [explosion/spacy-llm#458] [Ideas]
Thanks for your great work!

Could you kindly provide a simple tutorial on how can we integrate other open-source models on HF? Like integrating UniNER-7B for better NER ability.

Thanks again!

**Comment by rederyang** [comment]
Oh, maybe using langchain to interact with such models is a good practice?

**Comment by svlandeg** [comment]
Hi! The documentation about adding a new model is here: https://github.com/explosion/spacy-llm/tree/main/usage_examples#writing-your-own-model

For HF models specifically, have a look at the source folder https://github.com/explosion/spacy-llm/tree/main/spacy_llm/models/hf. It should be relatively easy to use any of those implementations as a template and adjust it to a new model for your specific purpose.
